firedrake/slate/slac/compiler.py

```diff
@@ -224,7 +229,23 @@ def generate_kernel_ast(builder, statements, declared_temps):
     # Now we wrap up the kernel ast as a PyOP2 kernel and include the
     # Eigen header files
     include_dirs = builder.include_dirs
-    include_dirs.extend(["%s/include/eigen3/" % d for d in PETSC_DIR])
+
+    def extract_eigen_include_dir(PETSC_ARCH):
+        try:
+            file = open("%s/lib/petsc/conf/petscvariables" % (PETSC_ARCH))
+        except:
+            raise FileNotFoundError("Could not find $PETSC_DIR/$PETSC_ARCH/lib/petsc/conf/petscvariables, where $PETSC_DIR/$PETSC_ARCH=%s." % PETSC_ARCH)
```

Comment:
Prefer context managers:

```
try:
   with open(...) as f:
       ...
except FileNotFoundError:
   ...
```

Reply:
Also prefer `os.path.join` over string interpolation.

Reply:
So just `os.path.join(PETSC_ARCH, "/lib/conf...")` or join on every sub-directory ?

Reply:
`os.path.join(PETSC_ARCH, "lib", "conf", ...)` (you can pass `*directories`)